src/ops/verify_proof.rs

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use super::AbiCall;
+use crate::call_context::{CallContext, Resolver};
+use crate::VMError;
+
+use canonical::Store;
+use dusk_plonk::prelude::*;
+use transfer_circuits::{
+    ExecuteCircuit, SendToContractObfuscatedCircuit,
+    SendToContractTransparentCircuit, WithdrawFromObfuscatedCircuit,
+};
+use wasmi::{RuntimeArgs, RuntimeValue, ValueType};
+
+pub struct ProofVerification;
```

Comment:
The naming convention is to reflect the actual host function, so call it `VerifyProof` (since the host function is called `verify_proof`).